tests/components/zha/common.py

```diff
@@ -237,3 +238,10 @@ async def async_test_rejoin(hass, zigpy_device, clusters, report_counts, ep_id=1
         assert cluster.bind.await_count == 1
         assert cluster.configure_reporting.call_count == reports
         assert cluster.configure_reporting.await_count == reports
+
+
+async def async_wait_for_updates(hass):
+    """Wait until all scheduled updates are executed."""
+    await asyncio.sleep(0)
+    await asyncio.sleep(0)
+    await hass.async_block_till_done()
```

Comment:
```suggestion
    await hass.async_block_till_done()
    await asyncio.sleep(0)
    await asyncio.sleep(0)
```

Reply:
shouldn't we keep async_block_till_done() as the last one? The tasks that were in the loop before the sleep, and that now were executed, may trigger other tasks in HA that will be tracked by async_block_till_done().

Reply:
have you tried it in this order and it was failing? The idea is that `async_block_till_done()` finishes all HA tasks and `asyncio.sleep()` should finish any other not tracked by the core. I just was thinking if we sleep before the block_till_done, then it is not guaranteed which tasks (tracked or untracked) it is going to schedule for running?

Reply:
No, I haven't tested the new order. But I've tested the previous order and it was passing.
Maybe we should add a async_block_till_done() at the end then? To handle possible new tracked tasks coming from the untracked ones?

Reply:
Yep -- you were right.